sdk/monitor/query/azlogs/fake/custom_server.go

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+package fake
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/monitor/query/azlogs"
+)
+
+func preferHeaderToQueryOptions(s string) *azlogs.QueryOptions {
```

Comment:
This is just deconstructing the header value and sticking the values back into their originating fields?

Reply:
yes, exactly. Need to insert it so that fakes works properly since I added some customization to the query options